Revert Section Z default value

smancill opened the pull request:
PR #2374 and #2385 broke Section Z on macOS and now looks like this:

Terminal, SF Mono:
<img width="150" alt="butchered-section-z-terminal-sf-mono" src="https://user-images.githubusercontent.com/238528/132932195-112905c7-52c0-48c1-9f25-5efd724efe7a.png">

Iterm2, Menlo:
<img width="150" alt="butchered-section-z-iterm2-sf-menlo" src="https://user-images.githubusercontent.com/238528/132932211-a0dd514d-9583-415f-9afd-d05d78d27163.png">

And they had problems: 

- It started from a non-issue, #2373 contradicts what is clearly described in the README about Section Z symbols.

- #2385 added extra colons to something that was clean and simple before and broke the section Z for the `:terminal` command, that now looks like this with extra space that cannot be removed:

  <img width="150" alt="Captura de Pantalla 2021-09-10 a la(s) 22 23 16" src="https://user-images.githubusercontent.com/238528/132932353-e3c94743-8418-4f83-8380-f4607b5b5176.png">

- Even if it makes sense to swap `☰` and `㏑`, it introduces a bunch of spacing issues that were not a problem with the original layout, as commented on #2374. And the proposed solution seems to have been "too hard to figure out a good default, introduce the configuration and let the user fix them", which I think is just wrong. The plugin should provide sane defaults, not making things look bad and let the user figure out how to fix it.

- Also, using `℅` for the default column number symbol is not a good choice on a 10-12 pt fixed font terminal. It looks like a `%`, and it is confusing.

This reverts the Section Z to what is was before those PRs, and as it is still described in the README. It has worked fine all this time, no reason to fix what is not broken.

Keep the added `colnr` configuration, which was a good nice addition.

Anyone that has a different preference of spaces, adding extra colons, using random unicode characters, they can still change all the parts to their like.

chrisbra commented:
I don't follow. #2373 was not caused by #2374 it what triggered the change and after some thinking it makes sense to use `㏑` for the current line number and `☰` for the max line. 

If we change this now, this will only add to the confusion. How about we find a good looking `colnr` symbol for macos? Or perhaps it's just the `☰` that makes problem on macos?

And for the terminal issue we just might just want to tweek section z for the terminal extension?

smancill commented:
> I don't follow. #2373 was not caused by #2374 it what triggered the change and after some thinking it makes sense to use `㏑` for the current line number and `☰` for the max line.
> 
> If we change this now, this will only add to the confusion. How about we find a good looking `colnr` symbol for macos? Or perhaps it's just the `☰` that makes problem on macos?

Sorry if I wasn't clear, I was trying to not sound too annoying. Yes, I know that #2374 was created to solve #2373, but my point is that the information presented in #2373 is wrong and it contradicts the readme. The symbols were not reversed, they were in the expected places as documented in the readme, and thus #2374 was unnecessary because there was no issue to fix.

But yes, it may be right that it makes more sense to swap `㏑` and `☰`. I am not arguing about that.

The problem is that by swapping them, now there is this whole spacing issue that becomes hard to get right, which was not a problem with the old default. Even the author [commented](https://github.com/vim-airline/vim-airline/pull/2374#issuecomment-835682010) that the spacing is now too complicated to deal with. And then #2385 introduced further spacing changes which broke the section for the terminal.

IMO those PR just added too much hassle for no reason and broke spacing, when the previous default has been working fine for years. A good default should just work, not be broken for a lot of users and expect them to configure the section to fix it. That's why I reverted it, but I kept the new configuration option which is the good thing that the PR added.

Anyway, it doesn't really matter to me too much. It's up to you to decide the default, and as long as the configuration options are available, I can fix any annoyances. (The default `colnr` should be changed, though. It's too similar to `%`.)

> And for the terminal issue we just might just want to tweek section z for the terminal extension?

But I do care about fixing this, because I can't right now. #2385 removed the `spc` after the `%` to add it in front of `linenr`, and if I remove the space from `linenr` to fix the terminal section z there will be no space between the `%` and the `linenr` in the main section z.

I have no energy to re-add the `spc` after the `%` with the current default, and tweak all the spaces to get it right. That's also why I reverted to the previous default. I know it was fine and it doesn't need any further changes.

chrisbra commented:
okay fair enough. I am still slightly negative on it, to avoid further confusion though. I'll leave this open for others to comment and if enough people feel this is justified I'll re-evaluate.
